Zend/zend_compile.c

```diff
@@ -8067,6 +8067,27 @@ static zend_string *zend_begin_method_decl(zend_op_array *op_array, zend_string
 		zend_error(E_COMPILE_WARNING, "Private methods cannot be final as they are never overridden by other classes");
 	}
 
+	if ((fn_flags & ZEND_ACC_ABSTRACT) &&
+		!(ce->ce_flags & (ZEND_ACC_EXPLICIT_ABSTRACT_CLASS|ZEND_ACC_TRAIT)) &&
+		!in_interface
+	) {
+		// Don't say that the class should be declared abstract if it is
+		// anonymous and can't be abstract
+		const char *msg;
+		if (ce->ce_flags & ZEND_ACC_ANON_CLASS) {
+			msg = "Anonymous class %s cannot contain abstract method %s::%s";
```

Comment:
This check also triggers for enums, which should need a separate error message.

Reply:
done, and test case added